test-suite/test-squid-conf.sh

```diff
@@ -97,27 +116,60 @@ errorLog="squid-stderr.log"
 $sbindir/squid -k parse -f $configFile 2> $errorLog
 result=$?
 
-if test -z "$failureRegex"
+# this is the value we return to our caller;
+# must be set by the code below using updateOutcome
+exitCode=""
+updateOutcome()
+{
+    local newOutcome="$1"
+    # never overwrite non-zero values (i.e. only overwrite null and zero)
+    if test -z "$exitCode" -o "$exitCode" = 0
+    then
+        exitCode="$newOutcome"
+    fi
+}
+
+if test -n "$messageRegex" && ! grep -q -E "$messageRegex" $errorLog
+then
+    echo "ERROR: Squid did not emit an expected message to stderr"
+    echo "    expected message regex: $messageRegex"
+    updateOutcome 1
+fi
+
+if test $expectFailure = no
 then
-    # a positive test does not require special $result interpretation
-    exit $?
+    if test "$result" -ne 0
+    then
+        echo "ERROR: Squid rejected valid $configFile; Squid exit code: $result"
+        updateOutcome $result
+    else
+        # stay silent about ordinary success
+        updateOutcome 0
+    fi
+else
+    if test "$result" -eq 0
+    then
+        echo "ERROR: Squid successfully parsed malformed $configFile instead of rejecting it"
+        updateOutcome 1
+    else
+        # stay silent about this expected failure (invisible in our output)
+        #echo "Squid rejected malformed $configFile as expected; Squid exit code: $result"
```

Comment:
I commented this old `echo` line out while refactoring this code because its result looks out of place in "make squid-conf-tests" output (which does not distinguish successful positive and successful negative tests). It was alarming (and distracting) a "make" output reader for no good reason IMO:

```
PASS: squid.conf test: squid.conf.default
Squid rejected malformed ./squidconf/bad-regex.conf as expected; Squid exit code: 1
PASS: squid.conf test: bad-regex.conf
PASS: squid.conf test: bug4832.conf
PASS: squid.conf test: empty.conf
PASS: squid.conf test: external_acl_type.conf
PASS: squid.conf test: mgr_passwd.conf
PASS: squid.conf test: regex.conf
PASS: squid.conf test: regressions-3.3.conf
PASS: squid.conf test: regressions-3.4.0.1.conf
PASS: squid.conf test: regressions-3.5.0.2.conf
PASS: squid.conf test: regressions-4.0.18.conf
PASS: squid.conf test: time_units.conf
```


If anybody insists on returning this message, I will undo this out-of-scope change without arguing.

```suggestion
        echo "Squid rejected malformed $configFile as expected; Squid exit code: $result"
```